source/service.go

```diff
@@ -169,7 +169,7 @@ func (sc *serviceSource) Endpoints() ([]*endpoint.Endpoint, error) {
 			continue
 		}
 
-		svcEndpoints := sc.endpoints(svc)
```

Comment:
I would use a different approach. Instead of passing an additional argument, I would use the service source. Same as used for publishInternal

Reply:
Thanks for the feedback, I did changed it a bit, but we can not use the service source for this because we want to use both annotation at the same time. I also changed the docs.